### PR TITLE
Google Oauth2 login implimented

### LINKED ISSUE #169: Login With Google account

### PR DESCRIPTION
**Pull request by jonathandavidlewis**
Closes #169 
Users can now create accounts and login with Google.

If a user created an account with a username and password:
They can add google login to their account simply by using it.

If a user created their account with google:
There is no capacity currently to allow them to login with a username and password.

When a user logs into an account using Google, it automatically grants them access to any existing account using their main account email address.
If you create an account with local strategy using another peron's email, they can access that account by logging in with google.
This strange caveat could be avoided by requiring email verification, but it's also probably a non issue.

### Note:
No password is stored with a new Google Auth user.
Password is no longer required.
On save of any new user in the db, validation runs to check there is EITHER a password OR a GoogleId. If there is neither, an error is thrown currently, so it is actually not possible to store a user without having credentials.

4d6e2a78-441c-46cd-b4d5-2998a7f352a6

### CODE REVIEW

File: client/src/modules/login/login.html [hunk 1]
```diff
@@ -40,5 +40,7 @@
     <div class="login-other-container-item flex flex-center">
       <a ui-sref="signup">Don't have an account? Sign up today</a>
     </div>
+    <br/>
```
Comment (AtreidesEnd) [inline client/src/modules/login/login.html]: don't use br's, use margin, more easily adjustable - also I think just a little bit more spacing than is there right now

File: db/models/user.js [hunk 2]
```diff
@@ -7,19 +7,27 @@ const Promise = require('bluebird');
 let userSchema = new Schema({
   name: {type: String, required: true},
   email: {type: String, required: true, unique: true},
-  password: {type: String, required: true},
-  profilePic: String
+  password: {type: String, required: false},
+  profilePic: String,
+  googleId: Number
```
Comment (AtreidesEnd) [inline db/models/user.js]: should we / can we enforce this as unique?

File: views/layouts/main.handlebars [hunk 3]
```diff
@@ -13,6 +13,9 @@
   <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-alpha.6/js/bootstrap.min.js" integrity="sha384-vBWWzlZJ8ea9aCX4pEW3rVHjgjt7zpkNpZk+02D9phzyeVkE+jo0ieGizqPLForn" crossorigin="anonymous"></script>
   <script src="//cdnjs.cloudflare.com/ajax/libs/moment.js/2.10.6/moment-with-locales.js"></script>
   <link href="//cdn.rawgit.com/indrimuska/angular-moment-picker/master/dist/angular-moment-picker.min.css" rel="stylesheet">
+  {{#if newToken}}
+    <script type="text/javascript">localStorage.setItem('awg_jwt', "{{newToken}}");</script>
+  {{/if}}
```
Comment (AtreidesEnd) [inline views/layouts/main.handlebars]: nice and simple... well done!

File: client/src/modules/login/signup.html [hunk 4]
```diff
@@ -68,5 +68,7 @@
     <div class="login-other-container-item flex flex-center">
       <a ui-sref="login">Already have an account? Login here</a>
     </div>
+    <br/>
+    <div class="login-other-container-item flex flex-center"><a href="./auth/google" target="_self"><img src="images/btn_google_signin_light_normal_web@2x.png" alt=""></a></div>
```
Comment (AtreidesEnd) [inline client/src/modules/login/signup.html]: let's make this match the signin buttons for size, for now something just looks a bit off with the styling, i think it's just the sizing proportion... but we'll see

**Comment by jonathandavidlewis** [comment]
thanks

**Pull request merged**